fix: cannot open vscode when use vscode-win32-x64 in Windows

BlackHole1 opened the pull request:
When executing `npm run gulp vscode-win32-x64` on the Windows platform, the `bundle-vscode` task is triggered. However, this task does not handle the path issues well, resulting in the contents of the `bootstrap-window.js` file not being injected into `workbench.js` and `processExplorer.js`.

The reason why VSCode does not encounter this issue is that the code compilation (compile-build) is performed on a Linux machine, while the Windows machine only executes the application build (vscode-win32-x64-ci will not trigger `bundle-vscode`).

PTAL @deepak1556

bpasero commented on build/gulpfile.vscode.js:
```diff
@@ -140,8 +140,8 @@ const bundleVSCodeTask = task.define('bundle-vscode', task.series(
 					if (
-						filePath.endsWith('vs/code/electron-sandbox/workbench/workbench.js') ||
-						filePath.endsWith('vs/code/electron-sandbox/processExplorer/processExplorer.js')) {
+						filePath.endsWith(path.join('vs', 'code', 'electron-sandbox', 'workbench', 'workbench.js')) ||
```
I think I would prefer us to convert backslash to slash and compare with the path as it was to ensure we are not running into other issues.

bpasero reviewed (changes requested):
There also seems to be another case here:

https://github.com/microsoft/vscode/blob/4ad5865e8d95df60baa9c22291f2be035de3d538/build/gulpfile.vscode.web.js#L93-L112

BlackHole1 committed:
fix: cannot open vscode when use vscode-win32-x64 in Windows

When executing `npm run gulp vscode-win32-x64` on the Windows platform, the `bundle-vscode` task is triggered. However, this task does not handle the path issues well, resulting in the contents of the `bootstrap-window.js` file not being injected into `workbench.js` and `processExplorer.js`.

The reason why VSCode does not encounter this issue is that the code compilation (compile-build) is performed on a Linux machine, while the Windows machine only executes the application build (vscode-win32-x64-ci will not trigger `bundle-vscode`).

Signed-off-by: Kevin Cui <bh@bugs.cc>

BlackHole1 commented:
@bpasero Done

bpasero commented:
Thanks, we are actually checking in the related JS file as well, so can you run a `npm run compile` in the `build` folder?

BlackHole1 commented:
@bpasero Completed.
